### PR TITLE
Added Dependency Download Feature To DataServices

### PR DESCRIPTION
**Pull request by SangeethKarunaratne**

### CODE REVIEW

**Comment by prabushi** [comment]
Related to https://github.com/wso2/product-ei/issues/4400

File: plugins/org.wso2.developerstudio.eclipse.artifact.dataservice/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -19,6 +20,7 @@ Import-Package: org.eclipse.jdt.core,
  org.wso2.developerstudio.eclipse.capp.maven.utils;version="[6.0.0,7.0.0)",
  org.wso2.developerstudio.eclipse.ds.presentation.ui,
  org.wso2.developerstudio.eclipse.general.project.dialogs;version="[6.0.0,7.0.0)",
+ org.wso2.developerstudio.eclipse.gmf.esb.presentation,
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.artifact.dataservice/META-INF/MANIFEST.MF]: Can we remove the esb dependency for the dss side?

Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.artifact.dataservice/META-INF/MANIFEST.MF]: Fixed

File: plugins/org.wso2.developerstudio.eclipse.artifact.dataservice/src/org/wso2/developerstudio/eclipse/artifact/dataservice/ui/wizard/NewDataSourceWizardPage.java [hunk 2]
```diff
@@ -43,1247 +64,1802 @@
...
+    private Composite cDataSourceContainer;
+    private Object[] cDataSourceComponents;
+    private Composite jndiContainer;
+    private Object[] jndiControls;
+    private Composite gSpreadSheetContainer;
+    private Object[] gSpreadSheetControls;
+    private Composite webDataSourceContainer;
+    private Object[] webDataSourceControls;
+    private File projectLocation;
+
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.artifact.dataservice/src/org/wso2/developerstudio/eclipse/artifact/dataservice/ui/wizard/NewDataSourceWizardPage.java]: Fix formatting issues.

**Pull request merged**